pymc_marketing/mmm/multidimensional.py

```diff
@@ -0,0 +1,1653 @@
+#   Copyright 2025 - 2025 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Media Mix Model class."""
+
+import itertools
+import json
+import warnings
+from typing import Any, Literal
+
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import pymc as pm
+import pytensor.tensor as pt
+import xarray as xr
+from pymc.model.fgraph import clone_model as cm
+from pymc.util import RandomState
+
+from pymc_marketing.mmm import HSGP
+from pymc_marketing.mmm.components.adstock import (
+    AdstockTransformation,
+    adstock_from_dict,
+)
+from pymc_marketing.mmm.components.saturation import (
+    SaturationTransformation,
+    saturation_from_dict,
+)
+from pymc_marketing.mmm.fourier import YearlyFourier
+from pymc_marketing.mmm.tvp import infer_time_index
+from pymc_marketing.model_builder import ModelBuilder, _handle_deprecate_pred_argument
+from pymc_marketing.model_config import parse_model_config
+from pymc_marketing.prior import Prior
+
+PYMC_MARKETING_ISSUE = "https://github.com/pymc-labs/pymc-marketing/issues/new"
+warning_msg = (
+    "This functionality is experimental and subject to change. "
+    "If you encounter any issues or have suggestions, please raise them at: "
+    f"{PYMC_MARKETING_ISSUE}"
+)
+warnings.warn(warning_msg, FutureWarning, stacklevel=1)
+
+
+class MMMPlotSuite:
```

Comment:
What if we move this class to another module? Maybe even `plot.py`?

Reply:
done. I fully deprecated the previous mmm/plot.py